geopandas/tests/test_geodataframe.py

```diff
@@ -1066,6 +1066,29 @@ def test_clip(self, naturalearth_cities, naturalearth_lowres):
         result = left.clip(south_america)
         assert_geodataframe_equal(result, expected)
 
+    def test_clip_sorting(self, naturalearth_cities, naturalearth_lowres):
+        """
+        Test sorting of geodataframe when clipping.
+        """
+        cities = read_file(naturalearth_cities)
+        world = read_file(naturalearth_lowres)
+        south_america = world[world["continent"] == "South America"]
+
+        unsorted_clipped_cities = geopandas.clip(cities, south_america, sort=False)
+        sorted_clipped_cities = geopandas.clip(cities, south_america, sort=True)
+
+        expected_sorted_index = pd.Index(
+            [55, 59, 62, 88, 101, 114, 122, 169, 181, 189, 210, 230, 236, 238, 239]
+        )
+
+        assert not (
+            sorted(unsorted_clipped_cities.index) == unsorted_clipped_cities.index
```

Comment:
In the docstring above you say the order with sort=False is random. Is that true, or is it deterministic but unsorted? I believe it is the latter but would be good to confirm, because if it's the former, this assertion can randomly fail if the random order happens to be the sorted order by chance.

Reply:
The latter. It is the order determined by GEOS which shall be deterministic.

Reply:
Okay great. I guess random is the technically not quite correct in the docstring, but I think it conveys the intent well enough.